Description: allow mail command to grab the mail title from dep3 formalism
 If the patch is formatted according to http://dep.debian.net/deps/dep3/ 
 then the mail subcommand will manage to extract a mail subject and
 description out of this formalism.
Forwarded: Sent 2014-01-18

---
 quilt/mail.in |   22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

Index: b/quilt/mail.in
===================================================================
--- a/quilt/mail.in
+++ b/quilt/mail.in
@@ -342,6 +342,18 @@
 			' $tmpdir/patch > $tmpdir/body
 		fi
 
+		# Does this patch have a Description: block?
+		if [ -z "$subject" ]
+		then
+			subject=$(formail -x Description: < $tmpdir/header |head -n 1)
+			if [ -n "$subject" ]
+			then
+				# Get all dep3 pseudo-headers (but the first line of the description, already used as a subject)
+				# into the body of the generated mail, plus the patch.
+				sed -e 's/^Description:.*$/Description:/' $tmpdir/patch > $tmpdir/body
+			fi
+		fi
+
 		# Does this patch have DESC // subject // EDESC?
 		if [ -z "$subject" ]
 		then
@@ -446,10 +458,14 @@
 	subject=$(formail -x Replace-Subject: < "$tmpdir/$patch" | join_lines)
 	if [ $status -ne 0 -o -z "$subject" ]
 	then
-		printf \
+		subject=$(formail -x Description: < "$tmpdir/$patch" | head -n 1)
+		if [ $status -ne 0 -o -z "$subject" ]
+		then
+			printf \
 $"Unable to extract a subject header from %s\n" "$(print_patch "$patch")" >&2
-		rm -rf $tmpdir
-		exit 1
+			rm -rf $tmpdir
+			exit 1
+		fi
 	fi
 	subjects[${#subjects[@]}]="$patch"$'\t'"$subject"
 done
